components/NumberInput.js

```diff
@@ -13,11 +13,22 @@ class NumberInput extends Component {
     this.setState(Object.assign({}, this.state, {value: parseInt(ev.target.value)}))
   }
 
-  blur(ev) {
+  _finishEditing() {
```

Comment:
They are actually totally exposed :).

They might not look like it but js has a real hard time making anything private.

I don't have a good link to js standard, but I'm happy to give feedback on the common js stuff. camelCase everywhere except maybe constants (CAPS) and classes (PascalCase). Prefer chaining functional methods (map/reduce/filter) to 'for' loops.

There's some ES2015/2016 stuff in here due to Babel (like destructuring assigment https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) that's a little "out there" for me too, but it's a good stretching exercise!